src/Desktop.Analyzers/Core/TypesShouldNotExtendCertainBaseTypes.cs

```diff
@@ -31,79 +34,114 @@ public abstract class TypesShouldNotExtendCertainBaseTypesAnalyzer : DiagnosticA
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182171.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(SystemXmlXmlDocumentRule, SystemApplicationExceptionRule, SystemCollectionsCollectionBaseRule, SystemCollectionsDictionaryBaseRule, SystemCollectionsQueueRule, SystemCollectionsReadOnlyCollectionBaseRule, SystemCollectionsSortedListRule, SystemCollectionsStackRule);
 
+        private static readonly Dictionary<string, DiagnosticDescriptor> s_badBaseTypesToDescriptor = new Dictionary<string, DiagnosticDescriptor>
+                                                    {
+                                                        {"System.ApplicationException", SystemApplicationExceptionRule},
+                                                        {"System.Xml.XmlDocument", SystemXmlXmlDocumentRule},
+                                                        {"System.Collections.CollectionBase", SystemCollectionsCollectionBaseRule},
+                                                        {"System.Collections.DictionaryBase", SystemCollectionsDictionaryBaseRule},
+                                                        {"System.Collections.Queue", SystemCollectionsQueueRule},
+                                                        {"System.Collections.ReadOnlyCollectionBase", SystemCollectionsReadOnlyCollectionBaseRule},
+                                                        {"System.Collections.SortedList", SystemCollectionsSortedListRule},
+                                                        {"System.Collections.Stack", SystemCollectionsStackRule},
+                                                    };
+
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var badBaseTypes = s_badBaseTypesToDescriptor.Keys
+                                .Select(bt => context.Compilation.GetTypeByMetadataName(bt))
+                                .Where(bt => bt != null)
+                                .ToImmutableHashSet();
+
+            context.RegisterSymbolAction((saContext) =>
```

Comment:
What I meant is if some type is not referenced by the target project, and doesn't exist or located in a separate assembly in .net core, then that type won't show up in the list. 

In this case, `System.ApplicationException` doesn'r exist in corefx, non-generic collections are in [`System.Collections.NonGeneric`](http://www.nuget.org/packages/System.Collections.NonGeneric/) and XmlDocument is in [`System.Xml.XmlDocument `](http://www.nuget.org/packages/System.Xml.XmlDocument)